DM-7066: Port to Python 3

timj opened the pull request.

timj committed:
Modernize Python to 2.7 coding standard

timj committed:
Ignore _build and doc/xml

timj committed:
PEP8 cleanups

timj committed:
Replace deprecated test asserts

assert_ and assertEquals

timj committed:
Replace filter(lambda) with list comprehension

timj committed:
Use open() instead of file()

timj committed:
Port to Python 3

* inherit from object
* "a in b" rather than "a in b.keys()"
* Use numbers.Integral rather than long, int

The numbers.Integral change requires we check
bool first because bool is an int.

timj committed:
Upgrade unittest tests to modern pytest compatibility

Also ignore pytest cache dir

PaulPrice commented on tests/testBlockTimingLog.py:
```diff
@@ -4,20 +4,17 @@
-import pdb                              # we may want to say pdb.set_trace()
-import os
-import sys
+# import pdb                              # we may want to say pdb.set_trace()
```
Kill.

PaulPrice commented on tests/testNoDebug.py:
```diff
@@ -28,17 +28,18 @@
 if len(testdir) > 0:
     sys.path.insert(0, testdir)
-import verifyOutput
+import verifyOutput  # noqa
```
Is used a couple of lines later; no need for `noqa`?

PaulPrice commented on .gitignore:
```diff
@@ -16,6 +16,7 @@ doc/*.tag
 doc/*.inc
 doc/doxygen.conf
 tests/.tests
+tests/.cache
```
Squash into previous commit?

timj replied:
Squashed into the py.test commit.

timj committed:
Add dependency python_future

timj committed:
Add script to run executable tests

Not all are included because some are triggered by verification
python tests already.

timj committed:
Rewrite output verification tests to use unittest

Significantly simplify the verifyOutput module so that it
makes use of modern features in python and uses assertListEqual.
The driver scripts now properly use unittest and can be run
with py.test.

The test output must still be canonicalized to account for field
attributes associated with messages appearing in random order.

timj committed:
Convert final two test files to unittest

testLog_1.py has a memory leak of 2 PropertyTest objects which
seems to be related to log.LogRec(). A tearDown to delete the
object does not help so for now the memory test is disabled.

timj commented:
@ktlim I've done all the requested changes but I also went crazy and converted the remaining tests so they can run with unittest.
